apps/passport-client/components/shared/PCDCardList.tsx

```diff
@@ -0,0 +1,279 @@
+import { PCD } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDTypeName } from "@pcd/semaphore-identity-pcd";
+import _ from "lodash";
+import { useCallback, useMemo, useState } from "react";
+import styled from "styled-components";
+import { usePCDCollection } from "../../src/appHooks";
+import { PCDCard } from "./PCDCard";
+
+type Sortable = {
+  name: string;
+  order: number;
+};
+type SortKey = keyof Sortable;
+type SortOption = {
+  render: () => JSX.Element;
+  key: SortKey;
+};
+const SORT_OPTIONS: SortOption[] = [
+  { render: () => <NameIcon />, key: "name" },
+  /**
+   * PCDs are naturally sorted by order of when it is added to the collection.
+   * The order of its sequence proximates the time it was added.
+   */
+  { render: () => <ClockIcon />, key: "order" }
+];
+type SortState = {
+  sortBy?: SortKey;
+  sortOrder?: "asc" | "desc";
+};
+
+export function PCDCardList({ pcds }: { pcds: PCD[] }) {
+  const mainPCDId = useMemo(() => {
+    if (pcds[0]?.type === SemaphoreIdentityPCDTypeName) {
+      return pcds[0]?.id;
+    }
+  }, [pcds]);
+
+  const pcdCollection = usePCDCollection();
+  const sortablePCDs = useMemo(
```

Comment:
based on my comment above on the `Sortable` type, this could be an `Array<Sortable<PCD>>`